src/main/java/data/QueryInformation.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.TimeZone;
+
+/** Class to hold query information. */
+public class QueryInformation {
+  /** Millieconds equivalent to 11:59:59 PM. */
+  private static final long MILLISECONDS_TO_END_OF_DAY = (24L * 60L * 60L * 1000L) - 1000L;
```

Comment:
nit: Would it change the results if only 1 ms was subtracted instead of 1000? I know right now the users can't specify any unit of time smaller than a day when uploading a receipt, but in case a receipt timestamp was at 11:59:59.999, it would technically be in the same day and should be returned in the query.